src/libstore/globals.hh

```diff
@@ -250,6 +250,17 @@ public:
         )",
         {"build-use-substitutes"}};
 
+    Setting<unsigned int> maxQuerySubstitutersJobs{
```

Comment:
Wouldn't it be better to just check if the substituter supports http/2 and then significantly increase this setting?

Reply:
The better solution is not to use threads at all here, which would obviate the need for any parallelism guessing. This is the trivial fix hopefully in time for Nix 2.4 that at least gives people an option to max out the connections. Any more complex solution should come later.